### PR TITLE
Tests for hashcode and ==

### PR DESCRIPTION
**Pull request by yoannLafore**
Hello,
This PR aims to provide tests for the immutability of dataclasses in our application. (#142)
To do so, it tests that the `hashcode` function and `==` operator are correctly overriden.

Those tests have been added for the dataclasses : `VoteFirestore`, `CreateAccountModel`, `NewPostState`.

Acceptance criterias:
- All classes that are immutable have their hashcode and == overrides tested

Thanks for reviewing this PR :+1:

### CODE REVIEW

File: test/models/database/vote/vote_firestore_test.dart [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import "package:flutter_test/flutter_test.dart";
+import "package:proxima/models/database/vote/vote_firestore.dart";
+
+void main() {
+  group("Vote Firestore testing", () {
+    test("hash overrides correctly", () {
+      final voteFirestore = VoteFirestore(hasUpvoted: true);
+      final hashCode = voteFirestore.hashCode;
+
+      expect(hashCode, isA<int>());
```
Comment (CHOOSEIT) [inline test/models/database/vote/vote_firestore_test.dart:10]: Why are you just testing that `hashCode` is an `int` and not that the expected hashcode is the same as the one computed from `voteFirestore`

Comment (yoannLafore) [reply test/models/database/vote/vote_firestore_test.dart]: You are completely right, this is a mistake.

**Review by CHOOSEIT — APPROVED** [review]
LGTM ! Great work thank you !

**Review by Aderfish — APPROVED** [review]
LGTM

**Pull request merged**